apps/backend/src/reconciliation/pos-reconciliation.service.ts

```diff
@@ -32,32 +32,35 @@ export class POSReconciliationService {
 
     // Basic EAGER! 1:1 matching
 
-    for (const payment of payments) {
+    for (const [pindex, payment] of payments.entries()) {
       // console.log(`Processing payment for ${payment.amount} - ${payment.id} - ${payment.timestamp}`)
-      for (const deposit of deposits) {
+      for (const [dindex, deposit] of deposits.entries()) {
         // console.log(`Processing deposit ${deposit.id} for ${deposit.transaction_amt} - ${deposit.timestamp}`)
 
         // TODO:make this a strategy pattern
         if (
           payment.amount === deposit.transaction_amt &&
           payment.method === deposit.card_vendor &&
+          deposit.match === false &&
           differenceInSeconds(payment.timestamp, deposit.timestamp) < 240
         ) {
+
+          // mutate the original array to use in next heurisitc match
```

Comment:
This similar problem will exist in cash totals as well, needs fix

Reply:
Could we implement the layered approach here, but invert it. So first we check 1:1 matching where difference in seconds is very small, then if no matches, continue to broaden the window until there are none yet left to be matched for the date?

Reply:
Okay, so there is no case in which a transaction and deposit could be delayed greater than 240 seconds?

Reply:
there could be but having a greater time span make more possibility for bad matches than good ones? this could be wrong but how I understand it now. we can verify with BAs.